pandas/tests/io/test_excel.py

```diff
@@ -881,8 +881,42 @@ def test_excel_multindex_roundtrip(self):
                             tm.assert_frame_equal(
                                 df, act, check_names=check_names)
 
-    def test_excel_oldindex_format(self):
-        # GH 4679
+    def test_excel_old_index_format(self):
+        # see gh-4679
+        filename = 'test_index_name_pre17' + self.ext
+        in_file = os.path.join(self.dirpath, filename)
+
+        # We detect headers to determine if index names exist, so
+        # that "index" name in the "names" version of the data will
+        # now be interpreted as rows that include null data.
```

Comment:
Yes, because the parameter got dropped.  That's because when we passed in `has_index_names`, the value of the parameter would only change to `True` if they were detected.